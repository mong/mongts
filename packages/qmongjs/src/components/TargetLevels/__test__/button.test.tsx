import LEGEND_BTN from "../button";
import { render } from "@testing-library/react";
import { it, expect } from "vitest";

const dummy_function = (n: string) => {
  n;
};

it("Button renders", async () => {
  const { container } = render(
    <LEGEND_BTN
      level="Høy måloppnåelse"
      legend_btn_class="high"
      update_show_level_filter={dummy_function}
      show_level_filter=""
    />,
  );
  expect(container).toMatchSnapshot();
  const { container: another } = render(
    <LEGEND_BTN
      level="Høy måloppnåelse"
      legend_btn_class="high"
      update_show_level_filter={dummy_function}
      show_level_filter="L"
    />,
  );
  expect(another).toEqual(container);
});

it("Button renders", async () => {
  const { container } = render(
    <LEGEND_BTN
      level="Høy måloppnåelse"
      legend_btn_class="high"
      update_show_level_filter={dummy_function}
      show_level_filter="H"
    />,
  );
  expect(container).toMatchSnapshot();
});

it("Button renders", async () => {
  const { container } = render(
    <LEGEND_BTN
      level="Bla bla bla"
      legend_btn_class="mamma"
      update_show_level_filter={dummy_function}
      show_level_filter="H"
    />,
  );
  expect(container).toMatchSnapshot();
});

it("Button renders", async () => {
  const { container } = render(
    <LEGEND_BTN
      level="Bla bla bla"
      legend_btn_class="anna"
      update_show_level_filter={dummy_function}
      show_level_filter="H"
    />,
  );
  expect(container).toMatchSnapshot();
});

it("Button renders", async () => {
  const { container } = render(
    <LEGEND_BTN
      level="Bla bla bla"
      legend_btn_class="anna"
      update_show_level_filter={dummy_function}
      show_level_filter="A"
    />,
  );
  expect(container).toMatchSnapshot();
});

it("Button renders", async () => {
  const { container } = render(
    <LEGEND_BTN
      level="Bla bla og bla"
      legend_btn_class="likes"
      update_show_level_filter={dummy_function}
      show_level_filter="L"
    />,
  );
  expect(container).toMatchSnapshot();
});
