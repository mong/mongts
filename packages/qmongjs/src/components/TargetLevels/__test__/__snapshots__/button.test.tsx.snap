// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<div>
  <button>
    <i
      style="padding-right: 0.2em;"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(59, 170, 52); font-size: 1.2rem;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </i>
    <div
      class="legend-text"
    >
      Høy måloppnåelse
    </div>
  </button>
</div>
`;

exports[`Button renders 2`] = `
<div>
  <button
    class="button_checked"
  >
    <i
      style="padding-right: 0.2em;"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(59, 170, 52); font-size: 1.2rem;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </i>
    <div
      class="legend-text"
    >
      Høy måloppnåelse
    </div>
  </button>
</div>
`;

exports[`Button renders 3`] = `
<div>
  <button>
    <i
      style="padding-right: 0.2em;"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(253, 156, 0); font-size: 1.2rem;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 256c0 136.966 111.033 248 248 248s248-111.034 248-248S392.966 8 256 8 8 119.033 8 256zm248 184V72c101.705 0 184 82.311 184 184 0 101.705-82.311 184-184 184z"
        />
      </svg>
    </i>
    <div
      class="legend-text"
    >
      Bla bla bla
    </div>
  </button>
</div>
`;

exports[`Button renders 4`] = `
<div>
  <button>
    <i
      style="padding-right: 0.2em;"
    />
    <div
      class="legend-text"
    >
      Bla bla bla
    </div>
  </button>
</div>
`;

exports[`Button renders 5`] = `
<div>
  <button
    class="button_checked"
  >
    <i
      style="padding-right: 0.2em;"
    />
    <div
      class="legend-text"
    >
      Bla bla bla
    </div>
  </button>
</div>
`;

exports[`Button renders 6`] = `
<div>
  <button
    class="button_checked"
  >
    <i
      style="padding-right: 0.2em;"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(227, 7, 19); font-size: 1.2rem;"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
        />
      </svg>
    </i>
    <div
      class="legend-text"
    >
      Bla bla og bla
    </div>
  </button>
</div>
`;
